website/docs/r/virtual_machine.html.markdown

```diff
@@ -134,97 +303,824 @@ routeable network interface within 5 minutes.
-  attempt to boot after creation.
-* `controller_type` - (Optional) Controller type to attach the disk to.  'scsi'
-  (the default), or 'ide' are supported options.
-* `keep_on_remove` - (Optional) Set to 'true' to not delete a disk on removal.
-
-<a id="cdrom"></a>
-## CDROM
-
-The `cdrom` block supports:
-
-* `datastore` - (Required) The name of the datastore where the disk image is
-  stored.
-* `path` - (Required) The absolute path to the image within the datastore.
-
-## Attributes Reference
-
-The following attributes are exported:
-
-* `id` - The instance ID.
-* `uuid` - The instance UUID.
-* `moid` - The instance MOID (Managed Object Reference ID).
-* `name` - See Argument Reference above.
-* `vcpu` - See Argument Reference above.
-* `memory` - See Argument Reference above.
-* `datacenter` - See Argument Reference above.
-* `network_interface/label` - See Argument Reference above.
-* `network_interface/ipv4_address` - See Argument Reference above.
-* `network_interface/ipv4_prefix_length` - See Argument Reference above.
-* `network_interface/ipv6_address` - Assigned static IPv6 address.
-* `network_interface/ipv6_prefix_length` - Prefix length of assigned static
-  IPv6 address.
-* `power_state` - The power state of the virtual machine. Can be one of
-  `poweredOff`, `poweredOn`, or `suspended`.
-
-~> **NOTE:** `power_state` is a pseudo-computed value which enforces
-Terraform's expectation that managed virtual machines are either powered on, or
-destroyed. You cannot edit this value to set a different expected power state.
+### CPU and memory options
+
+The following options control CPU and memory settings on the virtual machine:
+
+* `num_cpus` - (Optional) The number of virtual processors to assign to this
+  virtual machine. Default: `1`.
+* `num_cores_per_socket` - (Optional) The number of cores to distribute among
+  the CPUs in this virtual machine. If specified, the value supplied to
+  `num_cpus` must be evenly divisible by this value. Default: `1`.
+* `cpu_hot_add_enabled` - (Optional) Allow CPUs to be added to this virtual
+  machine while it is running.
+* `cpu_hot_remove_enabled` - (Optional) Allow CPUs to be removed to this
+  virtual machine while it is running.
+* `memory` - (Optional) The size of the virtual machine's memory, in MB.
+  Default: `1024` (1 GB).
+* `memory_hot_add_enabled` - (Optional) Allow memory to be added to this
+  virtual machine while it is running.
+
+~> **NOTE:** Certain CPU and memory hot-plug options are not available on every
+operating system. Check the [VMware Guest OS Compatibility
+Guide][vmware-docs-compat-guide] first to see what settings your guest
+operating system is eligible for. In addition, at least one `terraform apply`
+must be executed before being able to take advantage of CPU and memory hot-plug
+settings, so if you want the support, enable it as soon as possible.
+
+[vmware-docs-compat-guide]: http://partnerweb.vmware.com/comp_guide2/pdf/VMware_GOS_Compatibility_Guide.pdf
```

Comment:
is there an HTML version of this we could link to in order to be more specific?

Reply:
Could not find one, unfortunately. I think this one was actually linked from a KB article.